app/src/main/java/com/rkm/tasky/alarm/model/ReminderAlarm.kt

```diff
@@ -0,0 +1,45 @@
+package com.rkm.tasky.alarm.model
+
+import android.os.Parcel
+import android.os.Parcelable
+import com.rkm.tasky.feature.common.AgendaItemType
+import com.rkm.tasky.util.date.getCurrentDayInLocalDateTime
+import kotlinx.datetime.LocalDateTime
+import kotlinx.parcelize.Parceler
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+data class ReminderAlarm(
+    val id: String,
+    val title: String,
+    val desc: String? = null,
+    val time: LocalDateTime,
+    val type: AgendaItemType
+): Parcelable {
+    private companion object : Parceler<ReminderAlarm> {
+        override fun ReminderAlarm.write(parcel: Parcel, flags: Int) {
+            parcel.writeString(id)
+            parcel.writeString(title)
+            parcel.writeString(desc)
+            parcel.writeString(time.toString())
+            parcel.writeString(type.name)
+        }
```

Comment:
Two things here:

1. Parcelization is a data layer implementation detail which should not be associated with your domain models. In this case, a second class that is parcelabe would be better.
2. You will also benefit then from defining the parcelable model to have all primitive types. You wouldn’t need to implement custom serialization logic for more complex types. You also get to keep the easier to use, but complex types, in your domain layer, a win-win situation 😄 

```
@Parcelize
data class ReminderAlarmParcelable(
    val id: String,
    val title: String,
    val description: String?,
    val timeMillis: Long
) : Parcelable
```